src/main/java/io/fabric8/maven/docker/service/QueryService.java

```diff
@@ -52,7 +53,7 @@ public Container getContainer(String containerId) throws DockerAccessException {
      * @throws DockerAccessException in case of an remote error
      */
     public Container getContainerByName(final String containerName) throws DockerAccessException {
-        for (Container el : docker.listContainers(CONTAINER_LIMIT)) {
+        for (Container el : docker.listContainers(containerLimit)) {
```

Comment:
Having limit here is probably quite dangerous because one can easily miss the container with the given name returning a false result. The iteration is not optimal anyway, wonder whether there's an easier way to perform a 'fetchByName' ?